custom_components/frigate/media_source.py

```diff
@@ -365,24 +365,22 @@ def media_class(self) -> str:
         return self.frigate_media_type.media_class
 
 
-def _validate_year_month(
+def _validate_year_month_day(
     inst: RecordingIdentifier, attribute: attr.Attribute, data: str | None
 ) -> None:
     """Validate input."""
     if data:
-        year, month = data.split("-")
-        if int(year) < 0 or int(month) <= 0 or int(month) > 12:
+        year, month, day = data.split("-")
+        if (
+            int(year) < 0
+            or int(month) <= 0
+            or int(month) > 12
+            or int(day) <= 0
+            or int(day) > 31
```

Comment:
Since the number of days varies per month, I think a simpler and more robust validation is simply:

```
try:
  datetime.strptime(data, '%Y-%m-%d')`
except ValueError:
  raise ValueError("Invalid date in identifier: %s" % data)
```

You could even not bother with the try/except (since datetime will already raise a `ValueError`) but I think the above is a slightly clearer error message.